src/create-matcher.js

```diff
@@ -33,8 +33,9 @@ export function createMatcher (
 
     if (name) {
       const record = nameMap[name]
-      if (process.env.NODE_ENV !== 'production') {
-        warn(record, `Route with name '${name}' does not exist`)
+      if (process.env.NODE_ENV !== 'production' && !record) {
```

Comment:
behavior should be consistent between prod and dev, so we should move the env check to only apply to the warning.